pkgs/by-name/am/amazon-cloudwatch-agent/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  lib,
+  amazon-cloudwatch-agent,
+  buildGoModule,
+  coreutils,
+  fetchFromGitHub,
+  git,
+  gnumake,
+  nix-update-script,
+  testers,
+}:
+
+buildGoModule rec {
+  pname = "amazon-cloudwatch-agent";
+  version = "v1.300045.0";
+
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "amazon-cloudwatch-agent";
+    rev = "refs/tags/${version}";
+    hash = "sha256-zFlo1F+a/cRk+yfLLSEBXbCRvfK/Na4NyeuXPp9hvVU=";
+  };
+
+  vendorHash = "sha256-LsKPdoUFuZfVo6Cx5DbpE4Zfa92IkXN+x4D2NcXi2+E=";
+
+  # See the list in https://github.com/aws/amazon-cloudwatch-agent/blob/v1.300045.0/Makefile#L68-L77.
+  subPackages = [
+    "cmd/config-downloader"
+    "cmd/config-translator"
+    "cmd/amazon-cloudwatch-agent"
+    "cmd/start-amazon-cloudwatch-agent"
+    "cmd/amazon-cloudwatch-agent-config-wizard"
+  ];
+
+  # See https://github.com/aws/amazon-cloudwatch-agent/blob/v1.300045.0/Makefile#L57-L64.
+  #
+  # Needed for "amazon-cloudwatch-agent --version" to not show "Unknown".
+  postInstall = ''
+    mkdir -p $out/bin
+
+    cat > $out/bin/CWAGENT_VERSION << END_OF_FILE
+    ${version}
+    END_OF_FILE
+  '';
+
+  passthru = {
+    tests = {
+      version = testers.testVersion {
+        package = amazon-cloudwatch-agent;
+        command = "amazon-cloudwatch-agent --version";
+      };
+    };
+
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "CloudWatch Agent enables you to collect and export host-level metrics and logs on instances running Linux or Windows server.";
+    homepage = "https://github.com/aws/amazon-cloudwatch-agent";
+    license = lib.licenses.mit;
+    mainProgram = "amazon-cloudwatch-agent";
+    platforms = lib.platforms.unix;
+  };
```

Comment:
We don't use cloud watch agent at work as it doesn't work with journald. Instead we use fluent-bit for log shipping. This means that it will also not work out of the box on NixOS and will need more work

https://github.com/aws/amazon-cloudwatch-agent/issues/382

For this reason I personally don't have a lot of motivation to maintain this. But I can vibe check at work if we'd find it useful for the metrics collection part.

Reply:
It's a custom module. I should open a PR for it  perhaps.  I have a similar one in my homelab:

* https://github.com/arianvp/nixos-village/blob/main/nix/mixins/aws.nix
* https://github.com/arianvp/nixos-village/blob/main/nix/modules/fluent-bit.nix

Reply:
Nice! I'd be happy to be a maintainer.

I see this PR has been open for weeks; do you know what's blocking it?